Library/Homebrew/os/mac/ruby_mach.rb

```diff
@@ -0,0 +1,101 @@
+require "vendor/macho/macho"
+require "os/mac/architecture_list"
+
+module RubyMachO
+  # @private
+  def macho
+    @macho ||= begin
+      MachO.open(to_s)
+    end
+  end
+
+  # @private
+  def mach_data
+    @mach_data ||= begin
+      machos = []
+      mach_data = []
+
+      if MachO.fat_magic?(macho.magic)
+        machos = macho.machos
+      else
+        machos << macho
+      end
+
+      machos.each do |m|
+        arch = case m.cputype
+        when "CPU_TYPE_X86" then :i386
```

Comment:
This never matches because the string returned by `cputype` is `"CPU_TYPE_I386"` in this case, effectively breaking some functionality for IA32 (x86, whatever one likes to call it). See also woodruffw/ruby-macho#5.